src/sentry/models/groupopenperiod.py

```diff
@@ -195,11 +217,31 @@ def get_open_periods_for_group(
     return open_periods
 
 
+def create_open_period(group: Group, start_time: datetime) -> None:
+    if not features.has("organizations:issue-open-periods", group.project.organization):
+        return
+
+    latest_open_period = get_latest_open_period(group)
+    if latest_open_period and latest_open_period.date_ended is None:
+        logger.warning("Latest open period is not closed", extra={"group_id": group.id})
+        return
+
+    # There are some historical cases where we log multiple regressions for the same group,
+    # but we only want to create a new open period for the first regression
+    GroupOpenPeriod.objects.create(
+        group=group,
+        project=group.project,
+        date_started=start_time,
+        date_ended=None,
+        resolution_activity=None,
```

Comment:
outside of the scope of this pr, but curious if we should also be linking the occurrence that started the open period? (idk if we'll need it, but if we're tracking the resolution activity, the occurrence might come in handy too)